File: src/main/java/com/hohltier/decompiler/menu/file/File.java
package com.hohltier.decompiler.menu.file;

import com.hohltier.decompiler.services.LanguageService;
import javax.swing.*;
import java.awt.event.KeyEvent;

public class File extends JMenu {

    public File() {
        setText(LanguageService.getTranslation("file"));
        setMnemonic(KeyEvent.VK_F);

        add(new FileOpenFile());
        add(new JSeparator());
        add(new FileNewInstance());
        add(new FileExit());
    }

}

File: jvm/src/main/kotlin/moe/sota/decompiler/jvm/menus/file/File.kt
package moe.sota.decompiler.jvm.menus.file

import com.formdev.flatlaf.extras.components.FlatMenu
import java.awt.event.KeyEvent
import javax.swing.JSeparator
import moe.sota.decompiler.jvm.services.LanguageService

class File(
    fileOpenFile: FileOpenFile,
    fileCloseTab: FileCloseTab,
    fileNewInstance: FileNewInstance,
    fileExit: FileExit,
    languageService: LanguageService,
) : FlatMenu() {
    init {
        mnemonic = KeyEvent.VK_F
        text = languageService.getString("file")

        add(fileOpenFile)
        add(fileCloseTab)
        add(JSeparator())
        add(fileNewInstance)
        add(fileExit)
    }
}
